.evergreen/config.yml

```diff
@@ -1320,14 +1349,47 @@ tasks:
         script: |-
           set -o errexit
           ${PREPARE_SHELL}
-          git add .
-          git commit -m "add files"
-          export AZUREOIDC_DRIVERS_TAR_FILE=/tmp/mongo-rust-driver.tgz
-          git archive -o $AZUREOIDC_DRIVERS_TAR_FILE HEAD
-          export AZUREOIDC_TEST_CMD="PROJECT_DIRECTORY='.' ./.evergreen/install-dependencies.sh rust\
-                                     && PROJECT_DIRECTORY='.' .evergreen/install-dependencies.sh junit-dependencies\
-                                     && PROJECT_DIRECTORY='.' OIDC_ENV=azure OIDC=oidc  ./.evergreen/run-mongodb-oidc-test.sh"
-          bash $DRIVERS_TOOLS/.evergreen/auth_oidc/azure/run-driver-test.sh
+           ./.evergreen/install-dependencies.sh rust
+          source .cargo/env
+          export AZUREOIDC_DRIVERS_TAR_FILE=/tmp/mongo-rust-driver.tar
+          rustup default stable
+          export RUSTFLAGS="-C target-feature=+crt-static"
+          cargo test test::spec::oidc::azure --features azure-oidc --no-run --target x86_64-unknown-linux-gnu
+          rm ./target/x86_64-unknown-linux-gnu/debug/deps/*.d
+          tar -cf $AZUREOIDC_DRIVERS_TAR_FILE ./target/x86_64-unknown-linux-gnu/debug/deps/mongodb-*
+          tar -uf $AZUREOIDC_DRIVERS_TAR_FILE ./.evergreen
+          gzip $AZUREOIDC_DRIVERS_TAR_FILE
+          export AZUREOIDC_DRIVERS_TAR_FILE=/tmp/mongo-rust-driver.tar.gz
+          # Define the command to run on the azure VM.
+          # Ensure that we source the environment file created for us, set up any other variables we need,
+          # and then run our test suite on the vm.
+          export AZUREOIDC_TEST_CMD="PROJECT_DIRECTORY='.' OIDC_ENV=azure OIDC=oidc ./.evergreen/run-mongodb-oidc-test.sh"
+
+  - name: "oidc-auth-test-gcp-latest"
+    commands:
+    - command: shell.exec
+      params:
+        working_dir: src
+        shell: bash
+        script: |-
+          set -o errexit
+          ${PREPARE_SHELL}
+          ./.evergreen/install-dependencies.sh rust
+          source .cargo/env
+          export GCPOIDC_DRIVERS_TAR_FILE=/tmp/mongo-rust-driver.tar
+          rustup default stable
+          export RUSTFLAGS="-C target-feature=+crt-static"
+          cargo test test::spec::oidc::gcp --features gcp-oidc --no-run --target x86_64-unknown-linux-gnu
+          rm ./target/x86_64-unknown-linux-gnu/debug/deps/*.d
+          tar -cf $GCPOIDC_DRIVERS_TAR_FILE ./target/x86_64-unknown-linux-gnu/debug/deps/mongodb-*
+          tar -uf $GCPOIDC_DRIVERS_TAR_FILE ./.evergreen
+          gzip $GCPOIDC_DRIVERS_TAR_FILE
+          export GCPOIDC_DRIVERS_TAR_FILE=/tmp/mongo-rust-driver.tar.gz
+          # Define the command to run on the gcp VM.
+          # Ensure that we source the environment file created for us, set up any other variables we need,
+          # and then run our test suite on the vm.
```

Comment:
We could drop the `ls -la`, but I like leaving it here in case something does go wrongly at some point.